### PR TITLE
Feature purge collection endpoint

### PR DESCRIPTION
**Pull request by SyedSibtainRazvi**
# Etiquettes of raising a PR

PR reviews are a way to request peers to give feedback on the code that is going
to the mainline. This means, that PR should be in a reviewable state. Please ensure
that the following checkboxes are ticked. If any of them are not ticked, then
the PR is not ready for review

- [x] Did you test changes locally?
- [x] Did you check PR contents after opening the PR

# Related issue

- [Endpoint to purge collection cache](https://github.com/code-mancers/strapi-plugin-stellate/issues/4)

# Functionalities added or changed

- Added endpoint for refreshing cache

# Additional info
- https://strapi.io/blog/how-to-create-a-custom-api-endpoint-in-strapi
# Checklist

- [ ] Added/Updated Story
- [ ] Tests
- [ ] Translations
- [ ] Documentation
- [x] Uploaded screenshots
- [x] Verified on preview URL once

# Screenshots

## Page(s) if applicable

|   Device    |            Design             |           Implementation            |
| :---------: | :---------------------------: | :---------------------------------: |
| **Desktop** | ** Figma design screenshot ** ||

### CODE REVIEW

File: admin/src/components/ClearCache.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      setShowAlert(true);
+      setVariant("success");
+    } catch {
+      setLoading(false);
+      setShowAlert(true);
+      setVariant("danger");
+    }
+  };
+
+  const handleAlertClose = () => {
```
Comment (SujayPrabhu96) [inline admin/src/components/ClearCache.jsx]:
```suggestion
  const handleAlertClose = () => setShowAlert(false);
```

File: admin/src/components/ClearCache.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          ) : null}
+          {collection && (
+            <Box
+              padding={4}
+              hasRadius
+              background="neutral0"
+              shadow="tableShadow"
+            >
+              <Button fullWidth loading={loading} onClick={handleClick}>
+                {collection || "purge all cache"}
```
Comment (SujayPrabhu96) [inline admin/src/components/ClearCache.jsx]:
```suggestion
                {collection || "Purge all cache"}
```

File: admin/src/utils/api.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+export const refreshCollectionCache = async (query) => {
+  try {
+    const payload = query;
+    const response = await axios({
+      url: `stellate/cache/refreshcollectioncache`,
```
Comment (SujayPrabhu96) [inline admin/src/utils/api.js]:
```suggestion
      url: "stellate/cache/refreshcollection",
```

File: admin/src/utils/api.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    return response;
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+export const refreshCollectionCache = async (query) => {
+  try {
+    const payload = query;
```
Comment (SujayPrabhu96) [inline admin/src/utils/api.js]: No need to assign this to a variable. You can directly pass this to `data`

File: server/routes/index.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    method: "POST",
+    path: "/cache/refresh",
+    handler: "cacheController.refreshcache",
+    config: {
+      policies: [],
+    },
+  },
+  {
+    method: "POST",
+    path: "/cache/refreshcollectioncache",
```
Comment (SujayPrabhu96) [inline server/routes/index.js]:
```suggestion
    path: "/cache/refreshcollection",
```

File: server/routes/index.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    handler: "cacheController.refreshcollectioncache",
```
Comment (SujayPrabhu96) [inline server/routes/index.js]:
```suggestion
    handler: "cacheController.refreshcollection",
```

File: server/controllers/cache-controller.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      .cacheService();
+
+    if (response.error) {
+      return ctx.internalServerError(`Server error: ${response.error}`);
+    }
+
+    ctx.body = response;
+  },
+
+  async refreshcollectioncache(ctx) {
```
Comment (SujayPrabhu96) [inline server/controllers/cache-controller.js]:
```suggestion
  async refreshcollection(ctx) {
```

**Review by SujayPrabhu96 — CHANGES_REQUESTED** [review]

File: admin/src/components/ClearCache.jsx (hunk 1 above)
Comment (SyedSibtainRazvi) [reply admin/src/components/ClearCache.jsx]: Done

File: admin/src/components/ClearCache.jsx (hunk 2 above)
Comment (SyedSibtainRazvi) [reply admin/src/components/ClearCache.jsx]: Done

File: admin/src/utils/api.js (hunk 3 above)
Comment (SyedSibtainRazvi) [reply admin/src/utils/api.js]: Done

File: admin/src/utils/api.js (hunk 4 above)
Comment (SyedSibtainRazvi) [reply admin/src/utils/api.js]: Got it

File: server/routes/index.js (hunk 5 above)
Comment (SyedSibtainRazvi) [reply server/routes/index.js]: Done

File: server/routes/index.js (hunk 6 above)
Comment (SyedSibtainRazvi) [reply server/routes/index.js]: Updated

File: server/controllers/cache-controller.js (hunk 7 above)
Comment (SyedSibtainRazvi) [reply server/controllers/cache-controller.js]: Updated

File: .strapi-updater.json [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (SujayPrabhu96) [inline .strapi-updater.json]: This file can be added to .gitignore
https://forum.strapi.io/t/strapi-updater-json-tracking/2251

File: .gitignore [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+############################
+# Strapi
+############################
+
+.env
+license.txt
+exports
+.cache
+build
+.strapi-updater.json
```
Comment (SujayPrabhu96) [inline .gitignore:115]: Even though it is added to gitignore, why is it still adding to the PR? 🤔

File: admin/src/components/Alert.jsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable no-unused-vars */
+import React from "react";
+import { Alert as StrapiAlert } from "@strapi/design-system/Alert";
+
+export const Alert = ({ variant, onAlertClose }) => {
```
Comment (SujayPrabhu96) [inline admin/src/components/Alert.jsx]:
```suggestion
export const Alert = ({ variant, onAlertClose }) => ()
```

**Review by SujayPrabhu96 — CHANGES_REQUESTED** [review]

File: admin/src/components/Alert.jsx (hunk 10 above)
Comment (SyedSibtainRazvi) [reply admin/src/components/Alert.jsx]: Updated

File: .gitignore (hunk 9 above)
Comment (SyedSibtainRazvi) [reply .gitignore:115]: Let me check

**Review by SujayPrabhu96 — APPROVED** [review]

**Pull request merged**